prow/bump.sh

```diff
@@ -49,17 +53,77 @@ if ! ($SED --version 2>&1 | grep -q GNU); then
   exit 1
 fi
 
-new_version="v$(date -u '+%Y%m%d')-$(git describe --tags --always --dirty)"
-echo -e "version: $(color-version ${new_version})" >&2
-if [[ "${new_version}" == *-dirty ]]; then
-  echo -e "$(color-error ERROR): uncommitted changes to repo" >&2
-  echo "  Fix with git commit" >&2
+cd "$(dirname "${BASH_SOURCE}")"
+
+usage() {
+  echo "Usage: "$(basename "$0")" [--push || --list || vYYYYMMDD-deadbeef] [image subset...]" >&2
   exit 1
+}
+
+if [[ -n "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
+  echo "Detected GOOGLE_APPLICATION_CREDENTIALS, activating..." >&2
+  gcloud auth activate-service-account --key-file="$GOOGLE_APPLICATION_CREDENTIALS"
+  gcloud auth configure-docker
+  cmd="--push"
```

Comment:
eh if I have `GOOGLE_APPLICATION_CREDENTIALS` set locally and I `bump.sh --list` it will actually do push?

Reply:
No it will --push. Humans don't need to use that var. They are already authenticated. So I don't consider this a valid scenario.

Reply:
things like bootstrap and kubetest will all set this env for you if you run them locally... I feel like this should be explicit?

Reply:
The lifetime of the environment changes bootstrap and kubetest make do not survive the end of those processes. 

So once kubetest exists any environment changes it makes are erased. 

Is there a reason we need to worry about kubetest calling bump.sh?

Reply:
gah, you are right! I thought they will be leaked to the current shell...